include/rocksdb/db.h

```diff
@@ -1046,6 +1046,19 @@ class DB {
     return IngestExternalFile(DefaultColumnFamily(), external_files, options);
   }
 
+  // CreateColumnFamilyWithImport() will create a new column family with
+  // column_family_name and import external SST files specified in metadata into
+  // this column family.
+  // (1) External SST files can be created using SstFileWriter.
+  // (2) External SST files can be exported from a particular column family in
+  //     an existing DB.
+  // External files are deleted on a successful return.
+  virtual Status CreateColumnFamilyWithImport(
+      const ColumnFamilyOptions& options, const std::string& column_family_name,
+      const ImportColumnFamilyOptions& import_options,
+      const std::vector<LiveFileMetaData>& metadata,
+      ColumnFamilyHandle** handle) = 0;
```

Comment:
It's not a big deal but I'm thinking of whether there is a way to further prevent users from shooting their own shoe. The biggest risk of misuse of the API is that they set up a different comparator from the source data. If there a way to pass out the comparator or name of the comparator while exporting the data and pass it in here and do some sanity check?

Reply:
Good idea. Added comparator name in export and a check for that on import.